ocaml/xapi/cert_distrib.ml

```diff
@@ -4,42 +4,45 @@ module D = Debug.Make (struct let name = "cert_distrib" end)
 
 module XenAPI = Client.Client
 
-type existing_cert_strategy = Erase_old | Merge [@@deriving sexp]
-
 type 'a remote_call =
   API.ref_host -> (Rpc.call -> Rpc.response) -> API.ref_session -> 'a
 
 module WireProtocol = struct
   module Sexp = Sexplib.Sexp
   open Sexplib0.Sexp_conv
 
-  type cert_blob = string [@@deriving sexp]
+  (** [existing_cert_strategy] is used to determine how to treat existing
+      certs in /etc/stunnel/certs-pool
+      Erase_old => existing certs in the trusted certs dir will be removed
+      Merge     => merge incoming certs with certs in the trusted certs dir,
+                   resolving conflicts by taking the incoming cert *)
+  type existing_cert_strategy = Erase_old | Merge [@@deriving sexp]
+
+  type certificate = HostPoolCertificate | ApplianceCertificate
+  [@@deriving sexp]
 
-  type cert = {uuid: string; blob: cert_blob} [@@deriving sexp]
+  type certificate_file = {id: string; content: string} [@@deriving sexp]
```

Comment:
I am finding `id` a bit confusing - i believe it's actually just a path?

```suggestion
  type certificate_file = {path: string; content: string} [@@deriving sexp]
```

Reply:
not exactly, it's the "identity" of the certificate. For the internal certificates the id is a uuid and the path is `uuid ^ ".pem"`

Reply:
I see, in that case I think we can just store the filename (so we then don't need `CertificateProvider.filename_of_cert`)

Reply:
We still need to build the filename somehow from the uuid, and I think this is the right module that should know about the conversion. I'll try to think where can the conversion be done by the same module. (i.e. on Wireprotocol.certificale creation)